### PR TITLE
setMediationService(String) - Android ML SDK

### PR DESCRIPTION
**Pull request by RagnarR**
* Add in Facebook adapters `setMediationService` method
* Update adapters in 'mediation.adapter' folder

### CODE REVIEW

File: mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java [hunk 1]
```diff
@@ -35,6 +35,8 @@
 public abstract class PubnativeNetworkAdapter {
 
     private static String TAG = PubnativeNetworkAdapter.class.getSimpleName();
+    public static final String MEDIATION_SERVICE_NAME = "Pubnative ML";
```
Comment (cerberillo) [inline mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java]: This is something required in FAN only, so let's keep it in facebook base adapter instead

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java [hunk 2]
```diff
@@ -46,6 +46,7 @@
     public FacebookNetworkVideoAdapter(Map data) {
 
         super(data);
+        AdSettings.setMediationService(MEDIATION_SERVICE_NAME);
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java]: Can't we reuse the base adapter to set the mediation service instead of doing it in each individual adapter?

File: mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkVideoAdapter.java [hunk 3]
```diff
@@ -64,9 +64,17 @@ public void load(Context context) {
...
+            if (TextUtils.isEmpty(apiKey) || TextUtils.isEmpty(adSpaceName)) {
                 invokeLoadFail(PubnativeException.ADAPTER_MISSING_DATA);
             } else {
+                new FlurryAgent.Builder()
+                        .withLogEnabled(true)
+                        .withLogLevel(Log.VERBOSE)
+                        .build(context, apiKey);
+                if (!FlurryAgent.isSessionActive()) {
+                    FlurryAgent.onStartSession(context);
+                }
```
Comment (cerberillo) [inline mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkVideoAdapter.java]: how is this related to facebook?

File: mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkInterstitialAdapter.java [hunk 4]
```diff
@@ -63,9 +63,17 @@ public void load(Context context) {
...
+            if (TextUtils.isEmpty(apiKey) || TextUtils.isEmpty(adSpaceName)) {
                 invokeLoadFail(PubnativeException.ADAPTER_MISSING_DATA);
             } else {
+                new FlurryAgent.Builder()
+                        .withLogEnabled(true)
+                        .withLogLevel(Log.VERBOSE)
+                        .build(context, apiKey);
+                if (!FlurryAgent.isSessionActive()) {
+                    FlurryAgent.onStartSession(context);
+                }
```
Comment (cerberillo) [inline mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkInterstitialAdapter.java]: how is this related to facebook?

File: mediation.adaters/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        Log.v(TAG, "onLoggingImpression");
+        invokeImpressionConfirmed();
+    }
+
+    @Override
+    public void onAdClick(PubnativeAdModel model) {
+        Log.v(TAG, "onAdClicked");
+        invokeClick();
+    }
+}
```
Comment (cerberillo) [inline mediation.adaters/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java]: Since this PR is for Facebook setMediation, how is that this contains a video adapter?

We need to keep PR cleaned in case of a rollback I won't like to rollback something different than what it's stated

File: mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java (hunk 1 above)
Comment (RagnarR) [inline mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java]: I saw also the same method for AdMob Ads and decide to move it in base adapter

File: mediation.adaters/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java [hunk 6]
```diff
@@ -150,6 +154,15 @@ public View setNativeAd(View view) {
...
             mediaView.setNativeAd(mNativeAd);
+            Button btn = new Button(view.getContext());
+            RelativeLayout.LayoutParams btnParams = new RelativeLayout.LayoutParams(
+                    RelativeLayout.LayoutParams.WRAP_CONTENT, RelativeLayout.LayoutParams.WRAP_CONTENT);
+            btnParams.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM, RelativeLayout.TRUE);
+            btnParams.addRule(RelativeLayout.ALIGN_PARENT_RIGHT, RelativeLayout.TRUE);
+            btn.getBackground().setColorFilter(0xFF779C43, PorterDuff.Mode.MULTIPLY);
+            btn.setText(getCallToAction());
+            btn.setTextColor(Color.WHITE);
+            mediaView.addView(btn, btnParams);
```
Comment (cerberillo) [inline mediation.adaters/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java]: how is this related to facebook `setMediationService` value?

File: mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java (hunk 1 above)
Comment (cerberillo) [inline mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java]: let's keep it individual per network just in case a network requires a specific value

File: mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkVideoAdapter.java (hunk 3 above)
Comment (RagnarR) [inline mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkVideoAdapter.java]: @cerberillo It's only update old adapters

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java (hunk 2 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java]: If you mean `FacebookNetworkAdapter` - yes we can, but we will setup it in each method because we don't have a constructor for it.

File: mediation.adaters/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java (hunk 5 above)
Comment (RagnarR) [inline mediation.adaters/net/pubnative/mediation/adapter/network/FacebookNetworkVideoAdapter.java]: We forget update mediation.adapter folder in previous tasks

File: mediation.adaters/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java (hunk 6 above)
Comment (RagnarR) [inline mediation.adaters/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java]: Look at previous comments

File: mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java (hunk 1 above)
Comment (RagnarR) [inline mediation/src/main/java/net/pubnative/mediation/adapter/network/PubnativeNetworkAdapter.java]: ok - I will move it in `FacebookNativeAdModel` class

File: mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkVideoAdapter.java (hunk 3 above)
Comment (cerberillo) [inline mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkVideoAdapter.java]: Then this should go in a different PR

Comment (RagnarR) [inline mediation.adaters/net/pubnative/mediation/adapter/network/YahooNetworkVideoAdapter.java]: I already remove it

**Comment by cerberillo** [comment]
This is a very elegant solution, good job 👍

**Comment by cerberillo** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;uXSj2ycj&#x2F;1119-2-setmediationservice-string-android-ml-sdk"> (2) setMediationService(String) - Android ML SDK</a>
